refactor(server/main): remove getCoralInventory

googleoblivion opened the pull request:
## Description
availableCoral table is malformed and is unnecessary. Merges the function into the main event.
<!-- What does your pull request change? Why should it be merged? Does it fix an issue? -->

## Checklist

<!-- Put an x inside the [ ] to check an item, like so: [x] -->

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

googleoblivion committed:
refactor(server/main): remove unnecessary availableCoral table, combine into sellCoral event.

googleoblivion committed:
refactor(client/main): Prevent freezing with NPC.

googleoblivion committed:
refactor(server/main) Notify player and add not 0 check to count

Manason commented on server/main.lua:
```diff
@@ -8,48 +8,38 @@ local pickedUpCoralIndexes = {}
-        exports.ox_inventory:RemoveItem(src, coral.item, amount)
-        player.Functions.AddMoney('cash', math.ceil(reward / amount), 'sold-coral')
+        if count and count > 0 then
+            corals = corals - 1
```
```suggestion
            corals -= 1
```

Manason commented on server/main.lua:
```diff
@@ -8,48 +8,38 @@ local pickedUpCoralIndexes = {}
-        exports.ox_inventory:RemoveItem(src, coral.item, amount)
-        player.Functions.AddMoney('cash', math.ceil(reward / amount), 'sold-coral')
+        if count and count > 0 then
+            corals = corals - 1
```
Does modifying this value affect the for loop iteration? Or is the number of iterations set once and then the underlying variable change won't affect it?

googleoblivion replied:
@Manason can you clarify what you mean here? The corals -= 1 affecting for loop? corals is just the counter for the notify under it.

Manason replied:
The for loop is iterating from 1 to the value of corals, but then while executing, corals is being decremented. I'm wondering if this cuts the number of iterations short.

googleoblivion committed:
refactor(server/main): Proposed Changes

Co-authored-by: Manason <clanerp@gmail.com>

googleoblivion committed:
fix(server/main): Use static config count for iterating.

I think this was just oversight on my part anyways and not intended use for corals variable.

Manason reviewed (changes requested):
Instead of giving the money piecemeal, it may be better to track how much money should be given, and give it once. This also would make it so corals isn't needed to be tracked separately

googleoblivion committed:
refactor(server/main): Add payout variable and give money outside of for loop.

Could do this for items to remove also.
